src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -272,7 +276,8 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
 
             foreach (var formatter in formatters)
             {
-                if (formatter.CanWriteResult(formatterContext, contentType: null))
+                formatterContext.ContentType = null;
```

Comment:
Wouldn't it be a bug if `CanWriteResult()` updated the `formatterContext` then returned `false`? If I'm correct, would prefer a `Debug.Assert(formatterContext.ContentType == null);` at the bottom of this loop.